test/compilable/test12807.d

```diff
@@ -0,0 +1,16 @@
+void foo(T)(ref T t)
```

Comment:
please add the test case from Issue 18775 also.

Reply:
How do I add a testcase that ensures that a deprecation message does not get printed?

Reply:
add a comment like this : 

// REQUIRED_ARGS: -de

It tells the tester to treat deprecations as errors

Reply:
Added.